### DISCUSSION TITLE
How to set just the vnc resolution Width x Height?

### DISCUSSION THREAD

**Discussion by tinker2much** [gitbls/sdm#243] [General]
I just want to set the resolution, as I would in the raspi-config gui, or non-interactively: 

> sudo raspi-config nonint do_vnc_resolution <width>x<height>

The vnc plugin seems to do a lot more than this.

**Comment by gitbls** [comment]
Are you suggesting that the only thing you want to do is run that one VNC command?

If so, I'd suggest using the `runatboot` plugin to run a script with that command in it. Unfortunately, the plugin doesn't take a command line as the thing to do, so you need the script.

If this doesn't do it for you, can you provide more context?

**Comment by tinker2much** [comment]
I want to set the vnc resolution, so I thought of the vnc plugin. But that seems to be a larger beast. 

It seems like something the raspiconfig plugin could handle, but it's not yet set up to do so. (Although it looks like changing line 21 "vldargs=..." to add vnc_resolution would suffice to add that capability?)

**Comment by gitbls** [comment]
I'm waiting for things to settle out a bit on VNC before doing anything about this, since there are so many options: RPL adding wayvnc, connect, and then there's RealVNC and tigervnc/tightvnc. 

Which one do you want to set?

**Comment by tinker2much** [comment]
Make the change to sdm's raspiconfig plugin so it can do the equivalent of the following, As seen here in the RPi Docs: [VNC resolution](https://www.raspberrypi.com/documentation/computers/configuration.html#resolution-nonint):  

> sudo raspi-config nonint do_vnc_resolution <width>x<height>


NOTE: The actual raspi-config menu offers the following as choices: 
640x480
720x480
800x600
1024x768
1280x720
1280x1024
1600x1200                
1920x1080

In my particular case, I want the highest possible, so I would want to say to sdm: 

"--plugin raspiconfig:"vnc_resolution=1920x1080"

**Comment by gitbls** [comment]
Perfect. Too bad the author didn't put that in at the start, or at least mention it 😵

**Comment by gitbls** [comment]
Took a close look at this today. To do this with appropriate `sdm-fu` requires honoring the argument in raspiconfig, and also adding a bit of code to function `doconfig` in sdm-cparse. Done, and will be in the next release.
